Remotion/Data/DomainObjects.UnitTests/Mapping/TypeDefinitionCollectionFactoryTest.cs

```diff
@@ -0,0 +1,103 @@
+      _fakeClassDefinition = ClassDefinitionObjectMother.CreateClassDefinitionWithMixins(typeof(Order));
+    }
+
+    [Test]
+    public void CreateTypeDefinitionCollection ()
+    {
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Order), null))
+          .Returns(_fakeClassDefinition)
+          .Verifiable();
+
+      var typeDefinitions = _typeDefinitionCollectionFactory.CreateTypeDefinitionCollection(new[] { typeof(Order) });
+
+      _mappingObjectFactoryMock.Verify();
+
+      Assert.That(typeDefinitions, Is.EqualTo(new[] { _fakeClassDefinition }));
+    }
+
+    [Test]
+    public void CreateTypeDefinitionCollection_NoTypes ()
+    {
+      var typeDefinitions = _typeDefinitionCollectionFactory.CreateTypeDefinitionCollection(new Type[0]);
+
+      Assert.That(typeDefinitions, Is.Empty);
+    }
+
+    [Test]
+    public void CreateTypeDefinitionCollection_DerivedClassAreSet ()
+    {
+      var fakeClassDefinitionCompany = ClassDefinitionObjectMother.CreateClassDefinitionWithMixins(typeof(Company));
+      var fakeClassDefinitionPartner = ClassDefinitionObjectMother.CreateClassDefinition(classType: typeof(Partner), baseClass: fakeClassDefinitionCompany);
+      var fakeClassDefinitionCustomer = ClassDefinitionObjectMother.CreateClassDefinition(classType: typeof(Customer), baseClass: fakeClassDefinitionCompany);
+
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Order), null))
+          .Returns(_fakeClassDefinition)
+          .Verifiable();
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Company), null))
+          .Returns(fakeClassDefinitionCompany)
+          .Verifiable();
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Partner), fakeClassDefinitionCompany))
+          .Returns(fakeClassDefinitionPartner)
+          .Verifiable();
+      _mappingObjectFactoryMock
+          .Setup(mock => mock.CreateClassDefinition(typeof(Customer), fakeClassDefinitionCompany))
+          .Returns(fakeClassDefinitionCustomer)
+          .Verifiable();
+
+      var typeDefinitions =
+          _typeDefinitionCollectionFactory.CreateTypeDefinitionCollection(
+              new[] { typeof(Order), typeof(Company), typeof(Partner), typeof(Customer) });
+
+      _mappingObjectFactoryMock.Verify();
+
+      Assert.That(typeDefinitions.Length, Is.EqualTo(4));
+
+      var orderClassDefinition = (ClassDefinition)typeDefinitions.Single(cd => cd.Type == typeof(Order));
+      Assert.That(orderClassDefinition.DerivedClasses.Count, Is.EqualTo(0));
+
+      var companyClassDefinition = (ClassDefinition)typeDefinitions.Single(cd => cd.Type == typeof(Company));
+      Assert.That(companyClassDefinition.DerivedClasses, Is.EquivalentTo(new[] { fakeClassDefinitionPartner, fakeClassDefinitionCustomer }));
+    }
+  }
```

Comment:
missing the generic test

Btw: this kind of class is something I usally try to do test first (provided I know in advance that this is what I want. And at a certain point, you just know / take the plunge).

Reply:
What generic test?

Reply:
not covered right now: type.IsGeneric then type.GetOpenGenericType()
https://github.com/re-motion/Framework/pull/395/files#diff-bce87b3fdd90c2f34b1fc2e7f5db7a578f376e4b0691c52646341087664e629aR96
(unless I missed it)